transformer_engine/pytorch/module/layernorm_linear.py

```diff
@@ -355,7 +356,11 @@ def forward(
             ctx.requires_dgrad = inp.requires_grad
             ctx.normalization = normalization
             ctx.primary_weights_in_fp8 = primary_weights_in_fp8
-            ctx.is_first_module = FP8GlobalStateManager.is_first_fp8_module()
+            ctx.reduce_and_update_bwd_fp8_tensors = ctx.fp8
+            if requires_grad(inp, ln_weight, ln_bias, weight, bias):
+                ctx.reduce_and_update_bwd_fp8_tensors = (
+                    ctx.reduce_and_update_bwd_fp8_tensors or
+                    FP8GlobalStateManager.is_first_fp8_module())
```

Comment:
Won't this do the amax reduction after each FP8 module? Based on the PR description, it seems it should be:
```suggestion
            ctx.reduce_and_update_bwd_fp8_tensors = False
            if ctx.fp8 and requires_grad(inp, ln_weight, ln_bias, weight, bias):
                ctx.reduce_and_update_bwd_fp8_tensors = (
                    FP8GlobalStateManager.is_first_fp8_module()
                )
```
Similar changes may be needed for `LayerNormMLP` and `Linear`.